websites/management/commands/markdown_cleaning/parsing_utils_test.py

```diff
@@ -0,0 +1,81 @@
+import pytest
+
+from websites.management.commands.markdown_cleaning.shortcode_grammar import (
+    ShortcodeParser,
+    ShortcodeTag,
+)
+
+
+@pytest.mark.parametrize(
+    ["text", "expected"],
+    [
+        (
+            # Quote all the things
+            R"The quick {{< blargle doesnt_need_quotes >}} brown fox",
+            R'The quick {{< blargle "doesnt_need_quotes" >}} brown fox',
+        ),
+        (
+            # Pyparsing hates whitespace.
+            R"The quick {{< sup    2 >}} brown fox",
+            R'The quick {{< sup "2" >}} brown fox',
+        ),
+        (
+            # Example: Convert shortcode delimiter
+            R'Take {{< resource_link uuid "that {{< sup 123 >}}" >}}, regex.',
+            R'Take {{% resource_link "uuid" "that {{< sup 123 >}}" %}}, regex.',
```

Comment:
This was hard to handle with the regex approach. (Note that this is valid Hugo markdown. Just, {{< sup 123 >}} renders as plaintext, not as a shrotcode. When we did delimiter replacement with regexes, this replacement was wrong and caused the build to fail.

This still won't look good, but it won't cause things to blow up.